src/simulator/shape_template.h

```diff
@@ -0,0 +1,59 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    shape_template.h
+  \brief   C++ Interface: Abstract class for objects
+  \author  Yifeng Zhu, (C) 2020
+  \email   yifeng.zhu@utexas.edu
+*/
+//========================================================================
+
+#include <iostream>
+#include <vector>
+#include "eigen3/Eigen/Dense"
+#include "math/line2d.h"
+#include <cmath>
+#ifndef SHAPE_TEMPLATE_H
+#define SHAPE_TEMPLATE_H
+
+class ShapeTemplate{
```

Comment:
I don't love the name shape template, predominantly because the classes that implement it are not going to be just a bunch of individual shapes. Seems to me like most shapes will have the same behavior and just need to load their initial lines from somewhere as this is designed.

Reply:
How about `EntityBase` ? To indicate all entities inherit from it, and that this is the base class?

Reply:
This seems fine to me.